firmware/src/hal/sgx/test/secret_store/test_secret_store.c

```diff
@@ -0,0 +1,385 @@
+#include <assert.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "assert_utils.h"
+#include "secret_store.h"
+#include "mock_seal.h"
+#include "mock_ocall.h"
+
+// Error code for the sest API as defined in secret_store.c
+#define SEST_ERROR (0)
+// The maximum value that can be returned by sest_read
+#define MAX_SEST_READ_SIZE (255)
+
+void setup() {
+    mock_seal_init();
+    mock_ocall_init();
+    assert(sest_init());
+}
+
+void test_secret_exists_after_write() {
+    setup();
+    printf("Test secret exists after write...\n");
+
+    char* key = "key";
+    uint8_t secret[] = "secret";
+    // Ensure the secret doesn't exist before the write
+    assert(!sest_exists(key));
+    // Write the secret and ensure it now exists
+    assert(sest_write("key", secret, sizeof(secret)));
```

Comment:
Did you mean to use the variable here and on the following `sest_exists`?

Reply:
Nice catch, yes. It's supposed to be the variable, will change that.